stubs/pyjks/jks/jks.pyi

```diff
@@ -0,0 +1,107 @@
+from _typeshed import Incomplete
+from collections.abc import Iterable
+from typing import Any, NoReturn
+from typing_extensions import Literal, Self, TypeAlias
+
+from .util import AbstractKeystore, AbstractKeystoreEntry
+
+__version_info__: tuple[int, int, int] | tuple[int, int, int, str]
+__version__: str
+MAGIC_NUMBER_JKS: bytes
+MAGIC_NUMBER_JCEKS: bytes
+SIGNATURE_WHITENING: bytes
+
+_JksType: TypeAlias = Literal["jks", "jceks"]
+_CertType: TypeAlias = Literal["X.509"]
+_KeyFormat: TypeAlias = Literal["pkcs8", "rsa_raw"]
+
+class TrustedCertEntry(AbstractKeystoreEntry):
+    store_type: _JksType | None
+    type: _CertType | None
+    cert: bytes
+    def __init__(
+        self,
+        *,
+        type: str = ...,
+        cert: bytes = ...,
+        store_type: _JksType = ...,
+        alias: str = ...,
+        timestamp: int = ...,
+        **kwargs: Any,
+    ) -> None: ...
```

Comment:
This is pretty unsound -- we should either use `Literal` types in both the `__init__` signature and the annotations for the `__init__` parameters, or we should use `str` in both the `__init__` signature and the annotations for the `__init__` parameters. (The former is more precise but also more annoying for users of the stubs ;)

Reply:
You mean the `type` argument? Yeah, that was an oversight, changing to `_CertType`.

The `__init__` constructors in this library are wonky and not meant for public usage. Not sure if we can signal this in some way from type hints.

Reply:
> You mean the `type` argument?

Yup, that's what I meant :)



> The `__init__` constructors in this library are wonky and not meant for public usage. Not sure if we can signal this in some way from type hints.

Hmm, no, no way of signalling that (outside of putting a comment in the source code of the stubs for nobody to ever read 🙃)